### ISSUE TITLE
Support Non-Vector Inputs for Multiple Shooting

### ISSUE THREAD

**Issue by avik-pal** [SciML/BoundaryValueDiffEq.jl#174]
This needs to be handled similarly to MIRK, but someone needs to go in and do it.

**Comment by ChrisRackauckas** [comment]
Yes it will be more difficult for this library than for other SciML libraries. It will take some time to get a viable solution here.
